stdlib.sh

```diff
@@ -654,12 +840,443 @@ function __STDLIB_API_1_info() { # {{{
+# checkRes f "cannot chdir to '${dir}'"
+#
+# Normal output would be something like:
+#
+# [ EXEC  ] Chdir to '/usr/local/bin' [OK]
+#
+# If called with DEBUG=1, the output would be:
+#
+# [ EXEC  ] Chdir to '/usr/local/bin'
+# [ DEBUG ] About to run: 'cd /usr/local/bin'
+#  [OK]
+#
+# If called with PRETEND=1, the output would be:
+#
+# [ EXEC  ] Chdir to '/usr/local/bin'
+# [ DEBUG ] Pretend to run: 'cd /usr/local/bin'
+#  [OK]
+#
+###############################################################################
+
+# This function may be overridden
+#
+# The idea is to log a message (with optional new line at the end),
+# wait for exectuion to finish, then
+# log the overall result at the end of the same line
+# with one of the closing-tag function below.
+function __STDLIB_API_1_waitForRes() { # {{{
+	local message="${1:-}"
+	local -i newLine=${2:-0}
+
+
+	(( $std_PRETEND )) && message="(Pretend) ${message}"
+
+	(( $std_DEBUG == $STDLIB_DEBUG_NORMAL )) && newLine=1
+
+	(( !newLine ))  &&  std_DEBUG=1 __STDLIB_API_1_std::log -n "[  ${std_COLOUR_START_EXEC}EXEC${std_COLOUR_END}   ]" "${message} "
+	(( newLine ))  &&   std_DEBUG=1 __STDLIB_API_1_std::log "[  ${std_COLOUR_START_EXEC}EXEC${std_COLOUR_END}   ]" "${message} "
+
+	std_ERRNO=0
+	return 0
+
+} # __STDLIB_API_1_waitForRes # }}}
+
+
+# This function may be overridden
+#
+# Core closing tag function.
+# It takes car of displaying the closing tag and an optional message that match
+# the opening execution tag
+#
+function __STDLIB_API_1_showRes() { # {{{
+	local prefix="${1:-${std_LIB}}" ; shift
+	local data="${*:-}"
+	local pre=""
+	local post=""
+
+	if [[ -n "${data}" ]]; then
+		pre="- "
+		post=" "
+	fi
+
+	std_DEBUG=1 __STDLIB_API_1_std::log " [ ${prefix}" "${pre}${data}${post}]"
+
+	std_ERRNO=0
+	return 0
+} # __STDLIB_API_1_showRes # }}}
+
+
+# This function may be overridden
+#
+# Success closing tag
+#
+function __STDLIB_API_1_resOk() { # {{{
+	__STDLIB_API_1_showRes "${std_COLOUR_START_OK}OK${std_COLOUR_END}" "${@:-}"
+
+	# Don't stomp on std_ERRNO
+	return 0
+} # __STDLIB_API_1_resOk # }}}
+
+
+# This function may be overridden
+#
+# Warning closing tag
+#
+function __STDLIB_API_1_resWarning() { # {{{
+	__STDLIB_API_1_showRes "${std_COLOUR_START_WARNING}WARNING${std_COLOUR_END}" "${@:-}"
+
+	# Don't stomp on std_ERRNO
+	return 1
+} # __STDLIB_API_1_resWarning # }}}
+
+# This function may be overridden
+#
+# Error closing tag
+#
+function __STDLIB_API_1_resKo() { # {{{
+	__STDLIB_API_1_showRes "${std_COLOUR_START_FAIL}FAIL${std_COLOUR_END}" "${@:-}"
+
+	# Don't stomp on std_ERRNO
+	return 1
+} # __STDLIB_API_1_resKo # }}}
+
+
+# This function may be overridden
+#
+# Fatal closing tag
+#
+function __STDLIB_API_1_resFail() { # {{{
+	__STDLIB_API_1_showRes "${std_COLOUR_START_FAIL}FAIL${std_COLOUR_END}" "${@:-}"
+	die
+
+	# Dead code, die will immediately quit
+	return 1
+} # __STDLIB_API_1_resFail # }}}
+
+
+# This function may be overridden
+#
+# Logs tag, calls function, then dies
```

Comment:
I'm not 100% that this needs its own function, rather than allowing the user to perform these steps manually...
I could perhaps see a case for a function which takes a message, a function, and parameters and then executes the function, printing the message and aborting iff the function doesn't succeed?  This may be complex, though, as you'd want to be able to do:
```
succeedordie myfunc param1 param2 -- This is my failure message
```
... or:
```
succeedordie My function failed :( -- myfunc param1 param2
```